File: vortexdbg-api/src/main/java/com/vortexdbg/arm/backend/EventMemHook.java
package com.vortexdbg.arm.backend;

public interface EventMemHook extends Detachable {

    enum UnmappedType {
        Read,
        Write,
        Fetch
    }

    boolean hook(Backend backend, long address, int size, long value, Object user, UnmappedType unmappedType);

}

File: vortexdbg-api/src/main/kotlin/com/vortexdbg/arm/backend/EventMemHook.kt
package com.vortexdbg.arm.backend

interface EventMemHook : Detachable {

    enum class UnmappedType {
        Read,
        Write,
        Fetch
    }

    fun hook(backend: Backend, address: Long, size: Int, value: Long, user: Any?, unmappedType: UnmappedType): Boolean

}
